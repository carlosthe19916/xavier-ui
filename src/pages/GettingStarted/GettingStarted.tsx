import React, { createRef } from 'react';
import Dropzone from 'react-dropzone';
import {
    Button,
    EmptyStateBody,
    EmptyState,
    Title,
    Bullseye,
    EmptyStateVariant,
    TitleLevel
} from '@patternfly/react-core';
import { RouterGlobalProps } from '../../models/router';
import ProcessImprovementSvg from '../../PresentationalComponents/Icons/process-improvement.svg';
import './GettingStarted.scss';
import ReportsPage from '../../PresentationalComponents/ReportsPage/ReportsPage';
import { User } from '../../models';
import { Redirect } from 'react-router';

interface StateToProps {
    file: File | null,
    user: User
}

interface DispatchToProps {
    selectUploadFile: (file: File) => any;
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
};

class GettingStarted extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    onDrop = (files: File[]) => {
        this.props.selectUploadFile(files[0]);
        this.props.history.push('/reports/upload');
    };

    render() {
        const { user } = this.props;

        if (!user.firstTimeCreatingReports) {
            return <Redirect to="/reports" />;
        }

        const dropzoneRef: any = createRef();
        const openFileDialog = () => {
            if (dropzoneRef.current) {
                dropzoneRef.current.open();
            }
        };

        return (
            <ReportsPage>
                <Bullseye>
                    <EmptyState variant={ EmptyStateVariant.full }>
                        <div className="pf-c-empty-state__icon" style={ { opacity: 0.6 } }>
                            <ProcessImprovementSvg height="80px" />
                        </div>
                        <Title size="lg" headingLevel={ TitleLevel.h5 }>
                            Let Red Hat Migration Analytics suggest ways to optimize your environment
                        </Title>
                        <EmptyStateBody>
                            See how you can optimize your virtual environment by
                            uploading a Red Hat CloudForms generated inventory file
                            and then letting Red Hat Migration Analytics create
                            reports showing how you can save money and optimize
                            workloads - whether by migrating virtual machines or
                            migrating application to RHEL.
                        </EmptyStateBody>
                        <Dropzone
                            onDrop={ this.onDrop }
                            ref={ dropzoneRef }
                            noClick noKeyboard
                            multiple={ false }
                            accept={ [ 'application/zip', 'application/json' ] }
                        >
                            { ({ getRootProps, getInputProps }) => {
                                return (
                                    <div { ...getRootProps({ className: 'dropzone' }) }>
                                        <input { ...getInputProps() } />
                                        <Button
                                            type="button"
                                            variant="primary"
                                            onClick={ openFileDialog }>
                                            Get Started
                                        </Button>
                                    </div>
                                );
                            } }
                        </Dropzone>
                    </EmptyState>
                </Bullseye>
            </ReportsPage>
        );
    }
}

export default GettingStarted;
