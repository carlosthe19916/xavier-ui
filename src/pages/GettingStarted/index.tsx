import { GlobalState } from '../../models/state';
import GettingStarted from './GettingStarted';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

const mapStateToProps = (state: GlobalState) => {
    const {
        userState: {
            user
        }
    } = state;
    return {
        user
    };
};

const mapDispatchToProps = {
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(GettingStarted)
);
