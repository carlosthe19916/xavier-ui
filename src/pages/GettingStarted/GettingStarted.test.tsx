import { shallow } from "enzyme";
import GettingStarted from "./GettingStarted";
import Dropzone from "react-dropzone";

const props = {
  file: null,
  selectUploadFile: jest.fn(),
  user: {
    firstTimeCreatingReports: true
  },
  match: null,
  history: {
    push: jest.fn()
  },
  location: null
};

describe("GettingStarted", () => {

  it("expect to render", () => {
    const wrapper = shallow(
      <GettingStarted { ...props } />
    );
    expect(wrapper).toMatchSnapshot();
  });

});
