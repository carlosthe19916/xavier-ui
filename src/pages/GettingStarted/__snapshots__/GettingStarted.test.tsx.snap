// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted expect to render 1`] = `
<ReportsPage>
  <Bullseye
    className=""
    component="div"
  >
    <EmptyState
      variant="full"
    >
      <div
        className="pf-c-empty-state__icon"
        style={
          Object {
            "opacity": 0.6,
          }
        }
      >
        <Icon
          height="80px"
        />
      </div>
      <Title
        size="lg"
      >
        Let Red Hat Migration Analytics suggest ways to optimize your environment
      </Title>
      <EmptyStateBody>
        See how you can optimize your virtual environment by uploading a Red Hat CloudForms generated inventory file and then letting Red Hat Migration Analytics create reports showing how you can save money and optimize workloads - whether by migrating virtual machines or migrating application to RHEL.
      </EmptyStateBody>
      <Link
        className="pf-c-button pf-m-primary"
        replace={false}
        to="/reports/upload"
      >
        Get Started
      </Link>
    </EmptyState>
  </Bullseye>
</ReportsPage>
`;
